fastjsonschema/generator.py

```diff
@@ -399,13 +440,30 @@ def generate_properties(self):
                     prop_definition,
                     '{}_{}'.format(self._variable, key),
                     '{}.{}'.format(self._variable_name, key),
+                    context={
+                        "definition": self._definition,
+                        "key": key
+                    },
                 )
             if 'default' in prop_definition:
                 self.l('else: {variable}["{}"] = {}', key, repr(prop_definition['default']))
 
+        if 'patternProperties' in self._definition:
+            self.generate_func_code_block(
+                {'patternProperties': self._definition['patternProperties']},
+                self._variable,
+                '{}.{}'.format(self._variable_name, 'patternProperties'),
+            )
+            self.l('pattern_keys = set()')
+            with self.l('for key in {variable}_keys:'):
+                for pattern in self._definition['patternProperties'].keys():
+                    with self.l('if globals()["{}_re"].search(key):', pattern):
+                        self.l('pattern_keys.add(key)')
+            self.l('{variable}_keys -= pattern_keys')
```

Comment:
I don't understand yet why this block is here, why methods later are not enough.

Reply:
:-) Yep, those all properties are funny one. Will look at it in more detail on Monday.